exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/BasicFormatMatcher.java

```diff
@@ -72,7 +72,7 @@ public boolean supportDirectoryReads() {
 
   @Override
   public FormatSelection isReadable(DrillFileSystem fs, FileSelection selection) throws IOException {
-    if (isReadable(fs, selection.getFirstPath(fs))) {
+    if (isFileReadable(fs, selection.getFirstPath(fs))) {
```

Comment:
I think we need to check all of the selection for Drop, not just the first.  Otherwise we risk blowing away a directory which has a bunch of files, where only the first is readable.  Right?